### ISSUE TITLE
Compatibility with Python 3.12 or 3.13?

### ISSUE THREAD

**Issue by tianhuil** [pyinvoke/invoke#1015]
It looks like invoke only works up to Python 3.11.  Is there any interest in moving up to later version of python?  This is a great project and I haven't found a replacement for it!  Please maintain it.

**Comment by alwaysmpe** [comment]
What problem are you having? I'm using invoke on 3.13 without problems (although the type annotations need refining) so it at least partially works

**Comment by SnoopJ** [comment]
Not sure if there was a particular problem @tianhuil was having, but at the very least the [classifier list](https://github.com/pyinvoke/invoke/blob/506bf4e020c177a03cf4257a22969bad0845e4ee/setup.py#L70) and [test environments](https://github.com/pyinvoke/invoke/blob/506bf4e020c177a03cf4257a22969bad0845e4ee/.circleci/config.yml#L68) do not go past Python 3.11 and should probably be updated to include those newer versions.

**Comment by sandeep-gh** [comment]
i am having this issue with python 3.12: 
```
from patchwork.files import exists
  File "/home/kabira/Development/cloudworks-monallabs/patchwork/patchwork/files.py", line 7, in <module>
    from invoke.vendor import six
ImportError: cannot import name 'six' from 'invoke.vendor' (/home/adming/Execution/cloudworks/venv/lib/python3.12/site-packages/invoke/vendor/__init__.py)

```

**Comment by alwaysmpe** [comment]
@sandeep-gh that looks like a version problem, the main branch of invoke doesn't vendorize `six`, so you might need to get a different version of `invoke`

**Comment by sandeep-gh** [comment]
yes, my bad. Its the patchwork module that has not been updated and is causing the problem.

**Comment by schwaerz** [comment]
There is a rather old ticket about one thing that does not work when running invoke in Windows with python 3.12+: https://github.com/pyinvoke/invoke/issues/571

As soon as you are trying to use any binary installed from pypi, invoke won't find them anymore with 3.12+.

**Comment by jbbqqf** [comment]
Hi @tianhuil — drive-by triage check: I think this can be closed as resolved.

Python 3.12, 3.13 and 3.14 are all supported on current `main`:

- [`pyproject.toml`](https://github.com/pyinvoke/invoke/blob/main/pyproject.toml) declares `requires-python = ">=3.9"` and ships classifiers up through `Programming Language :: Python :: 3.14`.
- [`.circleci/config.yml`](https://github.com/pyinvoke/invoke/blob/main/.circleci/config.yml) runs the test matrix on `["3.10", "3.11", "3.12", "3.13", "3.14"]`.
- The most recent 3.14-specific regression (changed `fcntl`/`TIOCGWINSZ` behavior) was fixed in 2.1.4 via #1040 — see the [changelog](https://github.com/pyinvoke/invoke/blob/main/sites/www/changelog.rst).

If you're hitting a concrete failure on a specific 3.12/3.13/3.14 setup that's not already covered by a more recent issue (e.g. #1052 for one remaining 3.14 PTY case), it'd help to reopen with a minimal repro and the failing version. Otherwise this one looks safe to close.

---
_Drive-by triage from an outside contributor (data engineer GCP/Python, [jbbqqf](https://github.com/jbbqqf)). Comment drafted with Claude assistance; evidence and conclusion are mine. Happy to be told I missed something._
